### ISSUE TITLE
day 7: observation about processes: how long does it take to come out with a new Gridcoin version?

### ISSUE THREAD

**Issue by Erkan-Yilmaz** [Erkan-Yilmaz/GRC#39]
Please share your ideas/suggestions about process improvements that will help Gridcoin. Thank you!

<br>

August 15:
* I posted about security flaws being reported in Gridcoin
  * further info at #34 (and as can be seen there: the issues were known much earlier, but now we have it out open in the public)

August 21:
* Gridcoin dev @tomasbrod [applied an exploit](https://steemit.com/gridcoin/@erkan/who-staked-gridcoin-blocks-999990-to-1000010) at the "public" [1million-block](https://steemit.com/gridcoin/@erkan/brain-food-while-you-wait-for-gridcoin-s-1-million-block) event
  * see [this](https://github.com/gridcoin/Gridcoin-Research/issues/332#issuecomment-305290530) and [this](https://github.com/gridcoin/Gridcoin-Research/issues/345) (345 is already in released code)

<br>

Let's observe now:
* How much time will it take to get a (mandatory) Gridcoin version out (that fixes the issues from above)?
  * it seems that some fixes are already in the [staging branch](https://github.com/gridcoin/Gridcoin-Research/tree/staging)
* the thing also is: that exchanges need some time in advance to prepare for the mandatory
  * and especially Poloniex with its known delays (aka: not replying to support tickets) will be probably a critical factor
* let's be good-willing and take as counter August 15

<br>

related:
* #27: check the timeline there: took about 1 week to release a version for: 
  * "_new Gridcoin user? your wallet will not run :-(_"
  * the issue seems to have been to wait on Rob @gridcoin
  * What could be done elsewise in future? Give also @denravonska God powers for the repo and needed services (so e.g. he can build + release a Windows version)? Speeden up the autotools or Travis system (see milestones)?

<br>

see also:
* [Gridcoin dev milestones](https://steemit.com/gridcoin/@erkan/gridcoin-development-milestones-1-v8-stake-engine-security-performance-2-autotools-new-build-system) (August 15)
* more dev news in [here](https://steemit.com/gridcoin/@erkan/brain-food-while-you-wait-for-gridcoin-s-1-million-block)
* the [process](https://github.com/Erkan-Yilmaz/GRC/labels/process) category:
e.g. [You have an "out of sync" message in version 3.5.9.9 (but your blockchain is actually uptodate and fine) ?](https://github.com/Erkan-Yilmaz/GRC/issues/30)

<br>
<br>

---

* You like what I do? Take action [here](https://github.com/Erkan-Yilmaz/GRC/blob/master/thank_you.md).
* Did you know, you can also [SUBMIT content onto twitter account @GridcoinIRC ?](https://aboutgridcoin.blogspot.com/2017/08/did-you-know-you-can-also-submit.html)

**Comment by Erkan-Yilmaz** [comment]
see also the [process](https://github.com/Erkan-Yilmaz/GRC/labels/process) category

**Comment by gridcoin** [comment]
Incorrect- Although I dont mind being blamed when I am the bottleneck, and do take responsibility for creating the RSA weight exploit in the first place, this release process was not bottlenecked by me.  The current release process is already light, and frankly having me for the final release provides an extra set of eyes on the actual prod final release (as far as ensuring the client stakes, the cutover height etc).  

There is no way to release code directly to production without going through the process.  

Please elaborate on where you believe the process could have deployed the fix any sooner.

Things need to be pushed from test to staging to prod After they are tested before they can be released.

**Comment by Erkan-Yilmaz** [comment]
[4 days now](https://steemit.com/gridcoin/@erkan/4-days-after-gridcoin-s-new-stake-v8-came-into-effect-what-do-you-think-about-it-share-your-stories), no new Gridcoin version or [snapshot](https://github.com/gridcoin/Gridcoin-Research/issues/584)
